test/parallel_api/ranges/std_ranges_test.h

```diff
@@ -625,61 +630,102 @@ struct test_range_algo
     test_range_algo(std::array<int, 2> sizes) : n_serial(sizes[0]), n_parallel(sizes[1]) {}
 #endif
 
-    void test_view(auto view, auto algo, auto& checker, auto... args)
+    void test_view_host(auto view, auto algo, auto& checker, auto... args)
     {
         test<T, host_subrange<T>, mode, DataGen1, DataGen2>{}.host_policies(n_serial, n_parallel, algo, checker, view, std::identity{}, args...);
+    }
+
 #if TEST_DPCPP_BACKEND_PRESENT
-        test<T, usm_subrange<T>, mode, DataGen1, DataGen2>{}(n_device, TestUtils::get_dpcpp_test_policy<call_id>(), algo, checker, view, std::identity{}, args...);
-#endif //TEST_DPCPP_BACKEND_PRESENT
+    template <typename Policy>
+    void test_view_hetero(Policy&& exec, auto view, auto algo, auto& checker, auto... args)
+    {
+        test<T, usm_subrange<T>, mode, DataGen1, DataGen2>{}(n_device, CLONE_TEST_POLICY_IDX(exec, call_id), algo, checker, view, std::identity{}, args...);
     }
+#endif //TEST_DPCPP_BACKEND_PRESENT
+
+    struct subrange_view_fo
+    {
+        template <typename TArg>
+        auto operator()(TArg&& v) const
```

Comment:
A yet another style nitpick: note that Hungarian notation is not used in this file, I advice against it.